conformance/src/main/scala/za/co/absa/enceladus/conformance/DynamicConformanceJob.scala

```diff
@@ -69,9 +69,19 @@ object DynamicConformanceJob {
     // get the dataset definition
     val conformance = dao.getDataset(cmd.datasetName, cmd.datasetVersion)
     val dateTokens = cmd.reportDate.split("-")
+
+    val reportVersion = cmd.reportVersion match {
```

Comment:
Why not do this in the `CmdConfig`? Would lower the hassle of changing a lot of code

Reply:
I'm a little conflicted about this. On the one hand: 
1. `CmdConfig` is supposed to give you the provided command line args, if it starts fetching args from hadoop it might be breaking SRP a little bit.
1. `fsUtils` requires a `SparkSession` to read in the Hadoop configuration to check the latest version, so making `CmdConfig` dependent on a `SparkSession` seems like unwanted coupling to me.

On the other hand:
1. It would be nice to handle all the Spark Job arguments in a central place beforehand. As it stands they are being set in `CmdConfig` and `main()`.

Maybe a `CmdConfig` decorator which handles these additional responsibilities would solve this?

Reply:
I would just like to note that `FileSystemVersionUtils` and `SparkSession` are already in CmdConfig from previous PRs. Se this wouldn't add a dependency to it, just move a method.

Reply:
I see. In that case sounds like a good suggestion, since they're already inextricable coupled.

Reply:
So firstly there's an issue created for refactoring all these things into common functionality across std & conf. I believe this would be an ideal place for such changes and I would very much like to see all this refactored too.

Secondly, the version inference required the dataset publish path, which requires the DAO to be logged in etc. It's too much of a hassle right now.